webapp/src/Controller/API/JudgehostController.php

```diff
@@ -1166,6 +1168,121 @@ public function getFilesAction(
         };
     }
 
+    /**
+     * Get version commands for a given compile script.
+     */
+    #[IsGranted(new Expression("is_granted('ROLE_JURY') or is_granted('ROLE_JUDGEHOST')"))]
+    #[Rest\Get('/get_version_commands/{judgetaskid<\d+>}')]
+    public function getVersionCommands(string $judgetaskid): array
+    {
+        /** @var JudgeTask $judgeTask */
+        $judgeTask = $this->em->getRepository(JudgeTask::class)
+            ->findOneBy(['judgetaskid' => $judgetaskid]);
+        if (!$judgeTask) {
+            throw new BadRequestHttpException('Unknown judge task with id ' . $judgetaskid);
+        }
+
+        /** @var Submission $submission */
+        $submission = $this->em->getRepository(Submission::class)
+            ->findOneBy(['submitid' => $judgeTask->getSubmitid()]);
+        if (!$submission) {
+            throw new BadRequestHttpException('Unknown submission with submitid ' . $judgeTask->getSubmitid());
+        }
+
+        /** @var Language $language */
+        $language = $submission->getLanguage();
+
+
+        $ret = [];
+        if (!empty($language->getCompilerVersionCommand())) {
+            $ret['compiler_version_command'] = $language->getCompilerVersionCommand();
+        }
+        if (!empty($language->getRunnerVersionCommand())) {
+            $ret['runner_version_command'] = $language->getRunnerVersionCommand();
+        }
+        return $ret;
+    }
+
+    #[IsGranted(new Expression("is_granted('ROLE_JURY') or is_granted('ROLE_JUDGEHOST')"))]
+    #[Rest\Put('/check_versions/{judgetaskid}')]
```

Comment:
Would be nice (also for Mart 😛 ) if we document here with OA tags what the body must contain